.github/workflows/code.yml

```diff
@@ -79,7 +135,85 @@ jobs:
         # https://docs.github.com/en/actions/learn-github-actions/security-hardening-for-github-actions#using-third-party-actions
         uses: softprops/action-gh-release@2d72d869af3bf23602f9593a1e3fd739b80ac1eb  # v0.1.12
         with:
+          prerelease: ${{ contains(github.ref_name, 'a') || contains(github.ref_name, 'b') || contains(github.ref_name, 'rc') }}
           files: zips/*
           body: See [Changelog](CHANGELOG.rst) for more details
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  
+  # Container release
+  container-release:
+    needs: [build]
+    runs-on: ubuntu-latest
+    # if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')
+    steps:
+
+      # Git repositories
+      - name: Checkout Source
+        uses: actions/checkout@v2
+        with:
+          path: PandABlocks-rootfs
+
+      - name: Checkout rootfs builder
+        uses: actions/checkout@v2
+        with:
+          repository: dls-controls/rootfs
+          path: rootfs
+
+      - name: Checkout annotypes
+        uses: actions/checkout@v2
+        with:
+          repository: dls-controls/annotypes
+          path: annotypes
+
+      - name: Checkout pymalcolm
+        uses: actions/checkout@v2
+        with:
+          repository: dls-controls/pymalcolm
+          path: pymalcolm
+
+      - name: Checkout malcolmjs
+        uses: actions/checkout@v2
+        with:
+          repository: dls-controls/malcolmjs
+          path: malcolmjs
+
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+
+      - name: Log in to GitHub Docker Registry
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ghcr.io/pandablocks/pandablocks-dev-container
+          tags: |
+            type=ref,event=branch
+            type=ref,event=tag
+            type=raw,value=latest
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Rebuild and release image
+        uses: docker/build-push-action@v2
+        with:
+          context: . 
+          file: ./PandABlocks-rootfs/Dockerfile
```

Comment:
Have you worked out if it is quick to do the rebuild here?

If not maybe we could look at a different approach.

We really just need to do a docker push of the tarred artefact - but also would like to take advantage of Docker meta

Reply:
Last time I ran the CI, the rebuild and release took 4m 30s whereas the original build took 7m34. When we downloaded the artifact it took 2m53 for one job and 6 m for the other... The time before that, the rebuild and release took 3m 34s, the original build took 7m 21, and the artifact downloads took 4m 14s and 4m 35s. I would say it definitely seems faster than redoing the original build again. It's hard to tell if its quicker than downloading the artifact again....

Reply:
OK well then go with what you have.